src/geometry/interaction_groups.rs

```diff
@@ -7,12 +7,19 @@
 /// - The interaction groups filter.
 ///
 /// An interaction is allowed between two filters `a` and `b` when two conditions
-/// are met simultaneously:
+/// are met simultaneously for [`Self::test_and`] or individually for [`Self::test_or`]::
 /// - The groups membership of `a` has at least one bit set to `1` in common with the groups filter of `b`.
 /// - The groups membership of `b` has at least one bit set to `1` in common with the groups filter of `a`.
 ///
-/// In other words, interactions are allowed between two filter iff. the following condition is met:
+/// In other words, interactions are allowed between two filter iff. the following condition is met
+/// for [`Self::test_and`]:
 /// ```ignore
+/// ((self.0 >> 32) & rhs.0) != 0 && ((rhs.0 >> 32) & self.0) != 0
+/// ```
+/// or for [`Self::test_or`]:
+/// ```ignore
+/// ((self.0 >> 32) & rhs.0) != 0 || ((rhs.0 >> 32) & self.0) != 0
+/// ```
```

Comment:
The `iff.` followed by a condition with "or" is a bit difficult to follow, but that's probably me who is not used to `iff.`

Also, I'd probably use the user facing types (`InteractionTestMode::AND` rather than [`Self::test_and`]) and and/or link to [`Self::test`]: